src/Orc.Theming/Adorners/AsterixAdorner.cs

```diff
@@ -36,9 +38,31 @@ protected override void OnRender(DrawingContext drawingContext)
                       18, Brushes.Red, currentDpi
                    );
             }
-
+            DrawTextOnPosition(drawingContext);
+        }
+        private void DrawTextOnPosition(DrawingContext drawingContext)
+        {
             // Note: drawingContext.DrawGlyphRun can be used for better performance
-            drawingContext.DrawText(_cachedformattedText, new Point(drawRect.TopRight.X - _padding.Right, drawRect.TopRight.Y + _padding.Top));
+            var drawRect = new Rect(AdornedElement.DesiredSize);
+
+            switch (_positionCorner)
+            {
+                case PositionCorner.TopLeft:
+                    drawingContext.DrawText(_cachedformattedText, new Point(drawRect.TopLeft.X + _padding.Left, drawRect.TopLeft.Y + _padding.Top));
+                    break;
+                case PositionCorner.BottomRight:
+                    drawingContext.DrawText(_cachedformattedText, new Point(drawRect.BottomRight.X - _padding.Right, drawRect.BottomRight.Y - _padding.Bottom));
+                    break;
+                case PositionCorner.TopRight:
+                    drawingContext.DrawText(_cachedformattedText, new Point(drawRect.TopRight.X - _padding.Right, drawRect.TopRight.Y + _padding.Top));
+                    break;
+                case PositionCorner.BottomLeft:
+                    drawingContext.DrawText(_cachedformattedText, new Point(drawRect.BottomLeft.X - _padding.Right, drawRect.BottomLeft.Y + _padding.Bottom));
+                    break;
+                default:
+                    drawingContext.DrawText(_cachedformattedText, new Point(drawRect.TopRight.X - _padding.Right, drawRect.TopRight.Y + _padding.Top));
```

Comment:
Seems _padding.Left should be here. Please, double check every condition branch.